crates/chia-sdk-driver/src/primitives/datalayer/datastore_launcher.rs

```diff
@@ -0,0 +1,208 @@
+use chia_protocol::Bytes32;
+use chia_puzzles::{
+    nft::{NftStateLayerArgs, NFT_STATE_LAYER_PUZZLE_HASH},
+    EveProof, Proof,
+};
+use chia_sdk_types::Conditions;
+use clvm_traits::{FromClvm, ToClvm};
+use clvm_utils::{CurriedProgram, ToTreeHash, TreeHash};
+use clvmr::Allocator;
+
+use crate::{
+    DelegationLayerArgs, DriverError, Launcher, SpendContext, DL_METADATA_UPDATER_PUZZLE_HASH,
+};
+
+use super::{get_merkle_tree, DLLauncherKVList, DataStore, DataStoreInfo, DelegatedPuzzle};
+
+impl Launcher {
+    pub fn mint_datastore<M>(
+        self,
+        ctx: &mut SpendContext,
+        metadata: M,
+        owner_puzzle_hash: TreeHash,
+        delegated_puzzles: Vec<DelegatedPuzzle>,
+    ) -> Result<(Conditions, DataStore<M>), DriverError>
+    where
+        M: ToClvm<Allocator> + FromClvm<Allocator> + Clone + ToTreeHash,
```

Comment:
`ToClvm<Allocator> + ToTreeHash` is a bound we should avoid. You can use `ctx.allocator` and `tree_hash` to get the hash yourself here instead of using `ToTreeHash`